flutter_map/lib/src/layer/tile_layer.dart

```diff
@@ -395,5 +394,62 @@ class Coords<T extends num> extends Point<T> {
     return false;
   }
 
-  int get hashCode => hash3(x, y, z);
+  int get hashCode => _JenkinsSmiHash.hash3(x, y, z);
+}
+
+// Copyright (c) 2016, the Dart project authors.  Please see the FLUTTER_AUTHORS
+// file for details. All rights reserved. Use of this source code is governed by
+// a BSD-style license that can be found in the LICENSE file.
+//
+// Original commit in quiver-dart by Justin Fagnani <justinfagnani@google.com>
+// https://github.com/google/quiver-dart/commit/28e2805fc323e34ddd81b972745ee5b57cd2058e
+
+/// Jenkins hash function, optimized for small integers.
+///
+/// Static methods borrowed from sdk/lib/math/jenkins_smi_hash.dart.  Non-static
+/// methods are an enhancement for the "front_end" package.
+///
+/// Where performance is critical, use [hash2], [hash3], or [hash4], or the
+/// pattern `finish(combine(combine(...combine(0, a), b)..., z))`, where a..z
+/// are hash codes to be combined.
+///
+/// For ease of use, you may also use this pattern:
+/// `(new JenkinsSmiHash()..add(a)..add(b)....add(z)).hashCode`, where a..z are
+/// the sub-objects whose hashes should be combined.  This pattern performs the
+/// same operations as the performance critical variant, but allocates an extra
+/// object.
+class _JenkinsSmiHash {
+  /// Accumulates the hash code [value] into the running hash [hash].
+  static int combine(int hash, int value) {
```

Comment:
This makes me think that this will only work with integers, not doubles. Have you tested with doubles in strong mode?

Reply:
I've removed the `JenkinsSmiHash` altogether, since `dart:ui` exposes `hashValues()`